### ISSUE TITLE
Run-ALPipeline with App Name Change

### ISSUE THREAD

**Issue by MatthewMcCarthy7** [microsoft/navcontainerhelper#2587]
PLEASE DO NOT INCLUDE ANY PASSWORDS OR TOKENS IN YOUR ISSUE!!!

**Describe the issue**
Microsoft is making us change our App Name for AppSource, and when Run-ALPipeline runs it installs the previous version with the old name and then fails when installing the New Version with the new Name.  In the Run-AlPipeline script it checks that the Name, Publisher, and AppID all match to see if it has to do an upgrade, and since 2 of them do not, it does not perform the upgrade and fails since an app with the same ID already exists.  With Wave 1 2021 those can now be changed, https://docs.microsoft.com/en-us/dynamics365/business-central/dev-itpro/developer/devenv-app-identity.  When checking for installed apps it should just check against the AppID. 

**Full output of scripts**
[UpgradeLog.txt](https://github.com/microsoft/navcontainerhelper/files/9090145/UpgradeLog.txt)

```

**Screenshots**
I think this would solve the issue.

//BEFORE
    $installedApp = $false
    if ($installedApps | Where-Object { $_.Name -eq $appJson.Name -and $_.Publisher -eq $appJson.Publisher -and $_.AppId -eq $appJson.Id }) {
        $installedApp = $true
    }

// AFTER

    $installedApp = $false
    if ($installedApps | Where-Object { $_.AppId -eq $appJson.Id }) {
        $installedApp = $true
    }

```
**Additional context**
- does it happen all the time?
- did it use to work?

**Comment by freddydk** [comment]
I will look into this next week after my vacation

**Comment by freddydk** [comment]
Fix will be available in a pre-release in a few hours.

**Comment by freddydk** [comment]
Shipped in 4.0.3
